### ISSUE TITLE
Too bad validation accuracy

### ISSUE THREAD

**Issue by raeony** [fizyr/keras-retinanet#545]
hey, hgaiser
   I want to  transfer your  model in myself csv datasets(including various trademark billboard of total 60 classes). Everything seems normal in the training process, but whichever backbone(resnet152, restnet50) I choose and don't freeze any layer of it,  after 40 epochs iteration, it still get too bad validation accuracy in my validation datasets, just as the following picture:
![_20180703092451](https://user-images.githubusercontent.com/34431912/42193801-5cf4e68e-7ea3-11e8-98a6-c62f93c43ff1.png)
so, I just want to know whether is's nornal or not,  am I wasting time?
Looking forward to your answer, thanks a lot.

**Comment by yhenon** [comment]
It seems to be learning something, so I don't think you're wasting your time. A few things:
- your training loss is still quite high. You've only trained ~40k steps, which is really not that much (for reference, that would be 1/4 of an epoch on the COCO dataset)
- do you have enough training data?
- what is the trend of the validation set precision - is it upwards ?

**Comment by raeony** [comment]
@yhenon Thanks a lot for your reply!

- [ 1]  I have only 9000 pictures for training, of course each of them may contain one box or several boxes, so actually 13574 pictures can be feed into the model

- [ 2] yes,  the validation set precision seems upwards, but it's really too slow

- [3 ]  after considering what you are saying about steps, I adjust my steps_per_epoch to 8357 in order to fit my train set.  

- [ 4] again, thanks for your reply,  it's really helpful because it's possible that too small steps_per_epoch results in  a bad validation accuracy.

**Comment by tditt** [comment]
Maybe use debug.py to check if your annotations are being loaded correctly (you can let it output images of your training set with drawn annotation boxes).
I had great results training with pretrained weights, usually within the first 10 epochs (with 10000 steps per epoch). Tried resnet50, 101 and 152, trained to detect 1 class with a bit more than 100 images containing roughly 2000 class instances altogether.

**Comment by Taylor-Rose** [comment]
Hello, I wonder know how could I use this model to train my own dataset?

**Comment by hgaiser** [comment]
The answer of @yhenon and @tditt seems to be sufficient to answer the original question.

@Taylor-Rose I suggest to start with the README.

**Comment by Taylor-Rose** [comment]
@hgaiser I have never run the model like object dectation...I have read the README for several time. It says CSV files formation could be a good way, anyway thanks for your reply. I think I should try it and read more.

**Comment by Taylor-Rose** [comment]
I realized I didn't read the README carefully or maybe I was messed up by something. Now I get more clear than ever, thanks for your generous help @hgaiser @raeony

**Comment by Taylor-Rose** [comment]
@raeony Wow. Thanks for your help!

**Comment by Zweiyuan** [comment]
@raeony  hi, I saw your command.

> python keras-retinanet/bin/train.py --weights snapshots/resnet50_coco_best_v2.1.0.h5 --random-transform(for data-augmentation) csv annotations.csv classes.csv --val-annotations val-annotations(for validatation datasets)

I wonder know how to make data augmentation. 
It just need a commend **--random-transform** , or need to modify the script?
